### ISSUE TITLE
Improvements to simplify exponentiation

### ISSUE THREAD

**Issue by bobbinth** [maticnetwork/miden#411]
To be able to verify FRI proofs efficiently within the VM, we need perform some number of exponentiations in the field. Currently, this will be relatively expensive - i.e., probably a couple of hundred cycles, if not more. So, it would be nice if the assembly provided something like an `exp` instructions. This instruction would work as follows:
```
[b, a, ... ] -> [a^b, ... ]
```
This would be exponentiation in the field. Specifically, `a` would be treated as a field element, while `b` would be treated like an integer in the range [0, 2^64 - 2^32 + 1).

Implementing this instruction via a native VM operation would probably add too much complexity, but we could provide an operation which would simplify this task greatly. This operation would be similar to the `BINACC` operation we discussed in #200 and even started to implement in #308. However, it would be more general and would support exponentiation for any base and exponent. We can call this operation `EXPACC`. We can use this operation then to make `pow2` instructions more efficient, or we may even decide to get rid of `pow2` instructions altogether if using `exp` instruction can be used relatively easily instead.

To summarize, I think we should do the following:
* Add `EXPACC` operation to the VM.
* Use it to implement `exp` assembly instruction.
* Either rewrite `pow2` instructions to use `EXPACC` or remove them and replace their usage with `exp`.

**Comment by bobbinth** [comment]
With #415 we have the basic implementation completed. I think we need to do 2 more things to close this off:

* Implement parametrized version of `exp` instruction - e.g., `exp.u32`, `exp.123456` etc.
* Update (or remove) `pow2` instruction.

**Comment by bobbinth** [comment]
Closed by #415 and #420
